heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/filter/HeimdallDecorationFilter.java

```diff
@@ -249,24 +263,20 @@ protected HeimdallRoute getMatchingHeimdallRoute(String requestURI, String metho
 
         boolean auxMatch = false;
         for (Entry<String, ZuulRoute> entry : routeLocator.getAtomicRoutes().get().entrySet()) {
-            if (entry.getKey() != null) {
+            if (Objects.nonNull(entry.getKey())){
                 String pattern = entry.getKey();
                 if (this.pathMatcher.match(pattern, requestURI)) {
-
                     auxMatch = true;
-                    List<Credential> credentials = credentialRepository.findByPattern(pattern);
+                    List<Credential> credentials = credentialRepository.findByPattern(entry.getKey());
```

Comment:
So, you have returned variable, but is not using it. Please, use `pattern` variable.